zeus/api/resources/revision_styleviolations.py

```diff
@@ -0,0 +1,47 @@
+from flask import request
+from sqlalchemy.orm import contains_eager
+
+from zeus.constants import Severity
+from zeus.models import Job, StyleViolation, Revision
+from zeus.utils.builds import fetch_build_for_revision
+
+from .base_revision import BaseRevisionResource
+from ..schemas import StyleViolationSchema
+
+styleviolation_schema = StyleViolationSchema(many=True, strict=True)
+
+
+class RevisionStyleViolationsResource(BaseRevisionResource):
```

Comment:
This almost duplicates `BuildStyleViolationsResource`, the same way we have duplicated functionality in pretty much all `Build*Resource` and `Revision*Resource`. Probably makes sense to fix them all with one refactoring: e.g. move common code to some mixins and inject those in `BaseRevisionResource` and `BaseBuildResource`.

Reply:
Might also make sense to figure out how to do just generic aggregation.

e.g. TestCase already has this where we aggregate it to de-dupe